sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/CallbackRegistration.java

```diff
@@ -69,20 +70,23 @@ public InstrumentDescriptor getInstrumentDescriptor() {
   }
 
   void invokeCallback() {
-    // Return early if no storages are registered
-    if (noStoragesRegistered) {
-      return;
-    }
-    try {
-      callback.accept(measurement);
-    } catch (Throwable e) {
-      propagateIfFatal(e);
-      throttlingLogger.log(
-          Level.WARNING,
-          "An exception occurred invoking callback for instrument "
-              + instrumentDescriptor.getName()
-              + ".",
-          e);
+    // A callback MUST not be called concurrently
+    synchronized (lock) {
```

Comment:
What do you think about tightening it up so that the `synchronized(lock){ ... }` only surrounds the `callback.accept()` call? The exception handling wouldn't have to be within the lock would it?

Reply:
No strong preference. Seems reasonable.